kernel/environ.mli

```diff
@@ -46,8 +46,18 @@ type constant_key = Opaqueproof.opaque constant_body * (link_info ref * key)
 
 type mind_key = mutual_inductive_body * link_info ref
 
-type globals
-(** globals = constants + projections + inductive types + modules + module-types *)
+module Globals : sig
+  type t
+
+  type view =
+    { constants : constant_key Cmap_env.t
+    ; inductives : mind_key Mindmap_env.t
+    ; modules : module_body MPmap.t
+    ; modtypes : module_type_body MPmap.t
+    }
+
+  val view : t -> view
```

Comment:
Why a function instead of `type t = private view`?

Reply:
See the discussion above.

Reply:
I do not see.

Reply:
I was referring to this https://github.com/coq/coq/pull/8228#issuecomment-514552020 , but indeed, I am likely not understanding.

Reply:
Your first version was 
~~~ocaml
type globals = private { bla bla }
~~~
what is wanted is
~~~ocaml
type globals_internal = { bla bla }
type globals = private globals_internal
~~~

Reply:
What is the difference tho? I mean, that way you also expose the internal implementation.